### ISSUE TITLE
U-Net: Convolutional Networks for Biomedical Image Segmentation

### ISSUE THREAD

**Issue by HYUNMIN-HWANG** [yskim0/One-Day-One-Paper#30]
# U-Net: Convolutional Networks for Biomedical Image Segmentation
- https://arxiv.org/abs/1505.04597
- [Submitted on 18 May 2015]
- MICCAI 2015
---
## Summary
Biomedical Image를 augmentation한 후, Segmentation할 수 있는 U-Net 을 발표했다.

### 해당 논문에서 제안한 것들
- 적은 훈련 데이터 이미지로 segmentation에 있어 좋은 성능을 보여주었다.
![image](https://user-images.githubusercontent.com/70581043/128465723-e75e9ea1-be01-40ce-92a3-c9dd14aaf3d4.png)
- 전체적으로 CNN으로 이루어진 u-shaped architecture
> **1. contracting network**
> - two 3x3 convolutions (unpadded convolutions),
> - rectified linear unit (ReLU)
> - 2x2 max pooling- 사이즈를 줄여준다.
> - stride 2 for downsampling

> **2. expansive path**
> - 2x2 convolution ("up-convolution")
> - concatenation with the correspondingly cropped feature map from the contracting path,
> - two 3x3 convolutions,
> - ReLU
> - final layer a 1x1 convolution

> **3. overlap-tile strategy** 
> - 가장자리에 있는 비어있는 부분을 mirroring 방법으로 채워 넣었다.
> - 2x2 max-pooling operations

- fully connected layers 없음
- data augmentation 
> - elastic deformations방법을 사용함, 
> - 이 방법은 biomedical segmentation에서 tissue 혹은 현실적인 변형의 모습과 유사함 
> - using random displacement vectors on a coarse 3 by 3 grid.

- cell segmentation tasks : weighted loss를 사용함, background 라벨과 cell 라벨 간 큰 weight를 갖는다.

## Training
- stochastic gradient descent implementation
-  large batch size
- high momentum (0.99)
- energy function : a pixel-wise soft-max over the final feature map combined with the cross entropy loss function. (픽셀 x가 어느 k에 가까운지)
![image](https://user-images.githubusercontent.com/70581043/128466453-3036e372-c6c6-4ed7-89ce-77d07d2c7540.png)
- weight map : separation border is computed using morphological operations. (수식은 이해 못했음ㅠㅠ)
![image](https://user-images.githubusercontent.com/70581043/128466632-3473e779-ab0b-4f4d-a217-d70d616f84bb.png)

## Experiments
![image](https://user-images.githubusercontent.com/70581043/128467526-05b305a7-3357-42f5-bfee-835ca584b3e4.png)
- segmentation 결과 다른 모델들에 비해  warping error 가 작았음

![image](https://user-images.githubusercontent.com/70581043/128467769-ab245dcf-c852-4d6a-b4fc-d38f305da3d6.png)
- PhC-U373 데이터 셋과 DIC-HeLa 데이터 셋으로 cell segmentation한 결과, IOU가 가장 높게 나옴

**Comment by yskim0** [comment]
U-Net은 제로 패딩안쓰고 mirroring 방식으로 채워놓는 게 흥미로웠어요 ㅎㅎ 이게 biomedical 도메인에서 세포가 대칭이라서 그렇게 하는 게 더 효과적이다? 뭐 그런 얘기를 들었었는데 정말일까요? 제가 그쪽에 문외한이라 ㅎㅎ Biomedical 뿐만 아니라 general 도메인에도 잘 먹힌다는 것도 아주 흥미로웠습니다!~!

**Comment by HYUNMIN-HWANG** [comment]
@yskim0  저도 mirroring 방법 보고 참신하다고 생각했어요 !! 앞으로 다른 논문 볼 때 mirroring 사용하는 게 있는지 봐야겠어요 ㅎㅎ
